airbyte-integrations/connectors/source-okta/unit_tests/test_streams.py

```diff
@@ -22,67 +18,18 @@ def get_stream_by_name(stream_name: str, config: Mapping[str, Any]) -> Stream:
     return matches_by_name[0]
 
 
-class TestStatusCodes:
-    @pytest.mark.parametrize(
-        ("http_status", "should_retry"),
-        [
-            (HTTPStatus.OK, False),
-            (HTTPStatus.BAD_REQUEST, False),
-            (HTTPStatus.TOO_MANY_REQUESTS, True),
-            (HTTPStatus.INTERNAL_SERVER_ERROR, True),
-        ],
-    )
-    def test_should_retry(self, http_status, should_retry, url_base, start_date, requests_mock, oauth_config, api_url):
-        oauth_kwargs = {key: value for key, value in oauth_config.get("credentials").items() if key != "auth_type"}
-        oauth_kwargs["token_refresh_endpoint"] = f"{api_url}/oauth2/v1/token"
-        oauth_authentication_instance = CustomOauth2Authenticator(config=oauth_config, **oauth_kwargs, parameters=None)
-        oauth_authentication_instance.path = f"{api_url}/oauth2/v1/token"
-        assert isinstance(oauth_authentication_instance, CustomOauth2Authenticator)
-        source_okta = SourceOkta()
-        requests_mock.get(f"{api_url}/api/v1/users?limit=1", status_code=400, json={})
-        requests_mock.post(f"{api_url}/oauth2/v1/token", json={"access_token": "test_token", "expires_in": 948})
-        response_mock = MagicMock()
-        response_mock.status_code = http_status
-        stream = source_okta.streams(config=oauth_config)[0]
-        assert stream.retriever.requester._should_retry(response_mock) == should_retry
-
-
 class TestOktaStream:
     def test_okta_stream_request_params(self, oauth_config, url_base, start_date):
         stream = get_stream_by_name("custom_roles", config=oauth_config)
         inputs = {"stream_slice": None, "stream_state": None, "next_page_token": None}
         expected_params = {}
         assert stream.retriever.requester.get_request_params(**inputs) == expected_params
 
-    def test_okta_stream_backoff_time(self, url_base, start_date, oauth_config):
-        response_mock = requests.Response()
-        response_mock.status_code = 429
-        stream = get_stream_by_name("custom_roles", config=oauth_config)
-        expected_backoff_time = 60.0
-        assert stream.retriever.requester._backoff_time(response_mock) == expected_backoff_time
-
     def test_okta_stream_incremental_request_params(self, oauth_config, url_base, start_date):
         stream = get_stream_by_name("logs", config=oauth_config)
         inputs = {"stream_slice": None, "stream_state": None, "next_page_token": None}
         assert list(stream.retriever.requester.get_request_params(**inputs).keys())[0] == "since"
 
-    def test_incremental_okta_stream_backoff_time(self, oauth_config, url_base, start_date):
```

Comment:
Why delete a bunch of these tests?

Reply:
the reason I deleted those is that I was not able to find in the new sdk, the equivalent methods used such as the `should_retry`, or the `backoff_time`

So as I looked at the sdk code, the oktaSource declarative syntax uses As simpleRetriever with a Requester. I was not able to find equivalent methods

example
`assert stream.retriever.requester._should_retry(response_mock) == should_retry`